File: ide-common/src/main/java/org/digma/intellij/plugin/document/DocumentInfoChanged.java
package org.digma.intellij.plugin.document;

import com.intellij.psi.PsiFile;
import com.intellij.util.messages.Topic;

/**
 * Event fired when a document info has changed.
 * document info changes first when a file is opened. then when the file content changes
 * document info is computed again.
 */
public interface DocumentInfoChanged {

    Topic<DocumentInfoChanged> DOCUMENT_INFO_CHANGED_TOPIC = Topic.create("DOCUMENT_INFO_CHANGE_TOPIC", DocumentInfoChanged.class);

    void documentInfoChanged(PsiFile psiFile);
}

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/document/DocumentInfoChanged.kt
package org.digma.intellij.plugin.document

import com.intellij.openapi.vfs.VirtualFile
import com.intellij.util.messages.Topic
import org.digma.intellij.plugin.model.discovery.DocumentInfo


interface DocumentInfoChanged {


    companion object {
        @JvmField
        @Topic.ProjectLevel
        val DOCUMENT_INFO_CHANGED_TOPIC: Topic<DocumentInfoChanged> =
            Topic.create<DocumentInfoChanged>("DOCUMENT_INFO_CHANGE_TOPIC", DocumentInfoChanged::class.java)
    }


    fun documentInfoChanged(file: VirtualFile, documentInfo: DocumentInfo)
}
